### PR TITLE
[REFACTOR] Re-instate delete user functionality.

### PR DESCRIPTION
**Pull request by jenkoian**
With better support for multisite and VIP envs.

### CODE REVIEW

File: includes/class-gdpr-audit-log.php [hunk 1]
```diff
@@ -148,8 +151,26 @@ private static function email_mask( $email, $character = '-' ) {
 	 * @param  string $token   The 6 digit token the user gets on deletion.
 	 */
 	public static function export_log( $user_id, $token ) {
-		// see : https://github.com/trewknowledge/GDPR/blob/2.1.0/includes/class-gdpr-audit-log.php#L147.
-		throw new \DomainException('This function is not implemented.');
-	}
+		if (defined('VIP_GO_ENV') & false !== VIP_GO_ENV) {
+			// TODO: Fix this to work on VIP environments.
+			throw new \DomainException('This function is not implemented.');
+		}
```
Comment (jenkoian) [inline includes/class-gdpr-audit-log.php:157]: Will leave this for now but would like to test it out after as I think the code below will work for VIP envs as it provides a stream wrapper but want to just fix the issue at hand first then can revisit it. Need to test on a VIP env though so slow process.

File: includes/class-gdpr-audit-log.php [hunk 2]
```diff
@@ -104,9 +104,12 @@ public static function get_log( $email, $token = null ) {
 			$file_found   = file_exists( $path . $filename );
```
Comment (strayobject) [inline includes/class-gdpr-audit-log.php:104]: I think, that for VIP we should get the $wp_filesystem and use exists() method.

**Review by houdmont — APPROVED** [review]

File: includes/class-gdpr-audit-log.php (hunk 2 above)
Comment (strayobject) [reply includes/class-gdpr-audit-log.php]: Also, am I correct thinking that contents of `gdpr_logs/` are going to be publicly available and this is "security by obscurity"?

Comment (jenkoian) [reply includes/class-gdpr-audit-log.php:104]: So full VIP support of audit log is something I want to look at more next. I *think* this should work but would need to look into more closely.

**Review by strayobject — APPROVED** [review]

**Comment by jenkoian** [comment]
Travis failure is due to warnings on the VIP standard. I think the warnings are useful and thus don't want to suppress them, but I don't think this should stop merge.

**Pull request merged**